src/VerifyExamples/IntermediateAst.elm

```diff
@@ -0,0 +1,184 @@
+module VerifyExamples.IntermediateAst
+    exposing
+        ( IntermediateAst(..)
+        , Prefix(..)
+        , fromString
+        , group
+        , toString
+        )
+
+import List.Extra
+import Regex exposing (HowMany(..), Regex)
+import String
+
+
+type IntermediateAst
+    = MaybeExpression String
+    | Expression Prefix String
+    | Func String String
+    | NewLine
+
+
+type Prefix
+    = ArrowPrefix
+    | ImportPrefix
+    | TypePrefix
+
+
+toString : IntermediateAst -> String
+toString ast =
+    case ast of
+        MaybeExpression str ->
+            str
+
+        Expression _ str ->
+            str
+
+        Func _ str ->
+            str
+
+        NewLine ->
+            "\n"
+
+
+fromString : String -> List IntermediateAst
+fromString parsedComment =
+    commentLines parsedComment
+        |> List.filterMap toIntermediateAst
+
+
+commentLines : String -> List String
+commentLines str =
+    str
+        |> String.lines
+        |> List.concatMap splitOneLiners
+
+
+splitOneLiners : String -> List String
+splitOneLiners str =
+    let
+        breakIntoTwoLines a =
+            if Regex.contains expectationRegex a then
+                a
+            else
+                Regex.replace Regex.All arrowRegex (\_ -> "\n    --> ") a
+    in
+    str
+        |> breakIntoTwoLines
+        |> String.lines
+
+
+toIntermediateAst : String -> Maybe IntermediateAst
+toIntermediateAst =
+    oneOf
+        [ makeSyntaxRegex newLineRegex (\_ -> NewLine)
+        , makeSyntaxRegex importRegex (Expression ImportPrefix)
+        , makeSyntaxRegex typeRegex (Expression TypePrefix)
+        , makeSyntaxRegex expectationRegex (Expression ArrowPrefix)
+        , makeSyntaxRegex localFunctionRegex toFunctionExpression
+        , makeSyntaxRegex assertionRegex MaybeExpression
+        ]
+
+
+toFunctionExpression : String -> IntermediateAst
+toFunctionExpression str =
+    let
+        functionToTest : String
+        functionToTest =
+            str
+                |> Regex.find (AtMost 1) functionNameRegex
+                |> List.head
+                |> Maybe.andThen (.submatches >> List.head)
+                |> Maybe.andThen identity
+                |> Maybe.withDefault "no function name given!"
+    in
+    Func functionToTest str
```

Comment:
I think this would be better if either if: `functionToTest` was at the top level or: you chained it into `Func`

Reply:
agreed